### PR TITLE
spec: add scheduler entity

### PR DESCRIPTION
**Pull request by giuseppe**
extend the process struct to represent scheduling attributes for a process based on the sched_setattr(2) syscall.

I've been playing with crun and it is currently implemented as an annotation:

- https://github.com/containers/crun/pull/1164
- https://github.com/containers/crun/pull/1166

but I think this makes sense as part of the runtime-specs without the custom hack.

### CODE REVIEW

File: config.md
Comment (AkihiroSuda) [inline config.md:299]: Needs a complete list

File: specs-go/config.go
Comment (AkihiroSuda) [inline specs-go/config.go:43]: Probably this should be a pointer to differentiate zero from nil

Comment (giuseppe) [reply specs-go/config.go:43]: do we need that?  There is no unset for `sched_setattr`:

```
           struct sched_attr {
               u32 size;              /* Size of this structure */
               u32 sched_policy;      /* Policy (SCHED_*) */
               u64 sched_flags;       /* Flags */
               s32 sched_nice;        /* Nice value (SCHED_OTHER,
                                         SCHED_BATCH) */
               u32 sched_priority;    /* Static priority (SCHED_FIFO,
                                         SCHED_RR) */
               /* Remaining fields are for SCHED_DEADLINE */
               u64 sched_runtime;
               u64 sched_deadline;
               u64 sched_period;
           };
```

in this end this will be the value we set, unset will be more confusing

File: config.md
Comment (giuseppe) [reply config.md]: fixed

File: config.md
Comment (AkihiroSuda) [inline config.md:304]: Needs to be defined in specs-go too

File: specs-go/config.go
Comment (AkihiroSuda) [inline specs-go/config.go:827]:
```suggestion
	SchedOther LinuxSchedulerPolicy = "SCHED_OTHER"
```
To follow the current coding style:
https://github.com/opencontainers/runtime-spec/blob/main/specs-go/config.go#L677

Comment (giuseppe) [reply specs-go/config.go]: thanks, changed to camel case

**Review by AkihiroSuda — APPROVED** [review]

**Comment by rhatdan** [comment]
LGTM
@kolyshkin @mrunalp PTAL

**Review by mrunalp — APPROVED** [review]
LGTM

**Comment by utam0k** [comment]
Thanks, @giuseppe

File: specs-go/config.go
Comment (utam0k) [inline specs-go/config.go:43]: If you know, please let me know, does this apply to process groups?:pray: Or only one process?

File: config.md
Comment (utam0k) [inline config.md:304]: SCHED_DEADLINE seems to sometimes return an EBUSY error. Do you want the error numbers to follow these when the set fails?
https://man7.org/linux/man-pages/man2/sched_setattr.2.html

There seems to be some difficulty in using SCHED_DEADLINE with cpuset. can we consider removing SCHED_DEADLINE from support?
https://www.kernel.org/doc/Documentation/scheduler/sched-deadline.txt

Comment (giuseppe) [reply config.md:304]: I don't think it is required to support it, I am just listing the possible values for now (e.g. `SCHED_ISO` is not even implemented yet)

File: specs-go/config.go
Comment (giuseppe) [reply specs-go/config.go:43]: from my understanding that is only for one process or thread:

Comment (utam0k) [reply specs-go/config.go:43]: I agree with you. I thought the user would expect the nice value to be applied to all processes in the container? What do you think?

File: config.md
Comment (utam0k) [reply config.md:304]: I will leave it up to you to specify about it in the oci-spec, but there are some things that probably cannot be handled. I thought users might expect to be able to use them.

**Review by utam0k — COMMENTED** [review]
@giuseppe 
Sorry if I am causing trouble by leaving comments from the outside. If it doesn't seem necessary and the maintainer is satisfied, please ignore it and merge.

File: config.md
Comment (utam0k) [reply config.md:304]: How about specifying whether it applies to the container init process alone or to all processes in the container?

File: specs-go/config.go
Comment (utam0k) [reply specs-go/config.go:43]: How about specifying whether it applies to the container init process alone or to all processes in the container?

**Comment by giuseppe** [comment]
anything blocking or could we move forward with this PR?

File: schema/config-schema.json
Comment (thaJeztah) [inline schema/config-schema.json:155]: This should probably have an enum as well (`SCHED_OTHER`, `SCHED_FIFO` ...), and perhaps a definition (to match the Go type)

File: schema/config-schema.json
Comment (giuseppe) [reply schema/config-schema.json]: yes thanks for spotting it!

File: schema/config-schema.json
Comment (giuseppe) [reply schema/config-schema.json]: sure, fixed now

**Comment by giuseppe** [comment]
@thaJeztah thanks for the review.  Addressed your comments and pushed a new version

**Comment by giuseppe** [comment]
@thaJeztah does the new version look good to you?

File: specs-go/config.go
Comment (giuseppe) [reply specs-go/config.go]: sure, that seems like a good idea.  Fixed and pushed a new version

**Comment by giuseppe** [comment]
@thaJeztah could you PTAL?

**Review by thaJeztah — APPROVED** [review]
LGTM (thanks for the ping; sorry,  thought I already did 🙈 ❤️)

**Comment by thaJeztah** [comment]
Any more reviewers needed, or are we good to go? (forgot how many we want for such changes in this repository 🙈)

**Comment by thaJeztah** [comment]
Ah! Thanks @AkihiroSuda 😂

**Pull request merged**
